### PR TITLE
Manage organization "Teams"

### LINKED ISSUE #11073: Create a new Team
Feature request for organization account project in PyPI.

Description | The Owner or Manager of an Organization can create a new Team
-- | --
User value | The new Team can be used to assign specific permissions on Projects owned by the Organization
Acceptance criteria | Email notification to account Owners and Managers

### LINKED ISSUE #11074: Rename a Team
Feature request for organization account project in PyPI.

Description | The Owner or Manager of an Organization can rename a Team’s name
-- | --
User value | A team's name can be changed based on user preference. 
Acceptance criteria | Email notification to account Owners and Managers

### LINKED ISSUE #11075: Delete a Team
Feature request for organization account project in PyPI.

Description | The Owner or Manager of an Organization can delete a Team
-- | --
User value | If a Team is not required anymore, it can be deleted
Acceptance criteria | Email notification to account Owners and Managers

### LINKED ISSUE #11079: Add Member to a Team
Feature request for organization account project in PyPI.

Description | The Owner/Manager of an Organization account can add a Member to a Team
-- | --
User value | By adding Members to a Team, the same permission level can be set for the entire Team rather than for each User individually
Acceptance criteria | Email notification to account Owners, Managers and User

### LINKED ISSUE #11080: Remove Member from a Team
Feature request for organization account project in PyPI.
Description | The Owner/Manager of an Organization account can remove a Member from a Team
-- | --
User value | By removing Members from a Team, the permission assigned to the Member can be changed
Acceptance criteria | Email notification to account Owners, Managers and User. All permissions associated with the team are revoked for the User, unless they are granted directly. Owner/Manager removing the Member from the Team are prompted to confirm the action. This confirmation should include a notice of any permissions on Projects owned by the Organization that would still have effect after removal.

### LINKED ISSUE #11082: Assign manage/upload permission
Feature request for organization account project in PyPI.

Description | The Owner/Manager can assign manage/upload status to a team
-- | --
User value | The permission level can be set at a Team level
Acceptance criteria | Email notification to account Owners, Managers and Users

### LINKED ISSUE #11084: Dashboard to see Members, Teams, permission status, and projects
Feature request for organization account project in PyPI.

Description | The Owner/Manager can see a list of projects, teams associated with the project and Members in each Team
-- | --
User value | Easy to view each project, Team and Members and permission status

### PR DESCRIPTION
**Pull request by divbzero**
- Closes #11073.
- Closes #11074.
- Closes #11075.
- Closes #11079.
- Closes #11080.
- Closes #11082.
- Closes #11084.

### CODE REVIEW

**Comment by ewdurbin** [comment]
> @ewdurbin When I was adding journal entries/events, I came across a few inconsistencies in action/tag names that I tried to reconcile with commit [2264221](https://github.com/pypa/warehouse/pull/11665/commits/226422180cd0c471736f909591fb565d13905804). Please let me know —
> 
> 1. Do those changes make sense?
> 2. If yes, should I remove that commit from this PR and create a separate PR with those changes?

A separate PR is probably best.

File: warehouse/organizations/models.py
Comment (ewdurbin) [inline warehouse/organizations/models.py:442]: Since membership in teams can grant access, yes these should `HasEvents`

File: warehouse/organizations/models.py
Comment (ewdurbin) [inline warehouse/organizations/models.py:471]: members is probably better in my opinion.

File: warehouse/templates/manage/organization/teams.html
Comment (ewdurbin) [inline warehouse/templates/manage/organization/teams.html:39]: This seems to always be calculating as zero. I think it should be `team.users` unless the name is changed on the model to `members` as mentioned in `TODO`

**Review by ewdurbin — COMMENTED** [review]
A few minor things caught in review and some comments on TODO.

Overall this functions as expected. I believe user-testing will expose some specific user experience concerns, but I can't predict them at this time.

File: warehouse/organizations/models.py
Comment (divbzero) [reply warehouse/organizations/models.py]: Enabled `HasEvents` in commit [c7e2dcc](https://github.com/pypi/warehouse/pull/11665/commits/c7e2dcc84a6ea11e4947dba6fa929254b9037443).

File: warehouse/organizations/models.py
Comment (divbzero) [reply warehouse/organizations/models.py]: Renamed to `members` in commit [11733f6](https://github.com/pypi/warehouse/pull/11665/commits/11733f688a8839adcb6dc39a4183a481c78b9cf5).

File: warehouse/templates/manage/organization/teams.html
Comment (divbzero) [reply warehouse/templates/manage/organization/teams.html:39]: Renamed to `members` as mentioned above.

**Comment by divbzero** [comment]
> A separate PR is probably best.

Created separate PR #11779 to update project role journal entries and events.

File: warehouse/manage/forms.py
Comment (ewdurbin) [inline warehouse/manage/forms.py:43]: We should leave the labels for existing non-org projects as `Maintainer` and `Owner` until further notice, as merging this would change these labels and cause confusion for users.

**Review by ewdurbin — CHANGES_REQUESTED** [review]
The only outstanding change that needs to be addressed before this is mergable is that we should not change the current "Owner" and "Maintainer" role names for non-org projects.

Not only due to confusion in the UI, but also due to existing documentation, our own FAQ/help page, and 3rd party docs that reference these names.

**Comment by ewdurbin** [comment]
@divbzero note that this branch was rebased off of main to resolve translations merge conflicts that weren't auto-resolvable by GitHub

**Comment by divbzero** [comment]
@ewdurbin I’ve reverted the roles for individual users back to `Maintainer` and `Owner`, renamed permissions from `Admin` to `Administer`, and updated the project Collaborators page to clarify the distinction between roles and permissions.

File: warehouse/manage/forms.py
Comment (divbzero) [reply warehouse/manage/forms.py]: Fixed with commit [26bbc2e](https://github.com/pypi/warehouse/pull/11665/commits/26bbc2e824ce689cb4281287d0bec3e9dfb0b3fc).

**Comment by divbzero** [comment]
Fixed a few things that would have come up during user testing:
- Tweaked navigation for organizations and teams (commits [1c00802](https://github.com/pypi/warehouse/pull/11665/commits/1c00802f910f84257b7ed1833d2dc871422edecb) and [61e41b8](https://github.com/pypi/warehouse/pull/11665/commits/61e41b89bf8371310aa580fa0fa0cdf8b4aa85a1)).
- Fixed issue where teams with `Administer` permissions could’t actually administer the project (commit [3bce85c](https://github.com/pypi/warehouse/pull/11665/commits/3bce85c8f6fc364c1b41cbe0d84b524708a2a71f)).

Things should be cleaner now so users can focus on other usability issues.

@ewdurbin I think this pull request should be all set for merging. Please let me know if you spot anything else.

**Review by ewdurbin — APPROVED** [review]
This is working as expected and is ready for user testing.

**Pull request merged**
